table/block_based/filter_policy.cc

```diff
@@ -1135,6 +1234,14 @@ FilterBitsBuilder* BloomFilterPolicy::GetBuilderWithContext(
     const FilterBuildingContext& context) const {
   Mode cur = mode_;
   bool offm = context.table_options.optimize_filters_for_memory;
+  bool reserve_filter_construction_mem =
+      (context.table_options.reserve_table_builder_memory &&
+       context.table_options.block_cache);
+  std::shared_ptr<CacheReservationManager> cache_res_mgr;
```

Comment:
Considering that we can create CacheReservationManagers on the fly, it might make more sense to pass this one down as unique_ptr, so that it does not incur memory synchronization overhead associated with shared_ptr (for refcounting each time it's copied). Is that too much of a pain here?

Reply:
I thought of that but the issue is CacheReservationManagerHandle needs to share ownership to CacheReservationManager due to std::enable_shared_from_this<T>::shared_from_this by pointing to it.

Reply:
Oh right. The users of the Handles could pinky swear to close them before the CRM, but that's not a safe, friendly API.